common-utils/src/main/java/eu/arrowhead/common/model/ServiceModel.java

```diff
@@ -52,9 +52,6 @@ public Builder version(final String version) {
 
 		//-------------------------------------------------------------------------------------------------
 		public Builder serviceInterface(final InterfaceModel serviceInterface) {
-			if (interfaces == null) {
-				interfaces = new ArrayList<>();
-			}
```

Comment:
That is not good change. If someone set the interface list to null (with the other method), then use this method, you get an NPE. 

I think this is related to my comment in the other PR: "If you do this, you have to modify the Builder's serviceInterface method to not do anything if the input is null." But I see now, that the null input was already handled here, I just didn't notice when I reviewed your code.

So I think you should just reversed this change.

Reply:
Yeah but if I reverse it, than a null interface added after a previous non null interface will reset the interface list to an empty list. I think the `serviceInterfaces(final List<InterfaceModel> interfaces)` should be than modified not to do anything or reset to an empty list  if the input list is null.

Reply:
discussed via Skype